packages/foam-vscode/src/dated-notes.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  Selection,
+  Uri,
+  window,
+  workspace,
+  WorkspaceConfiguration
+} from "vscode";
+import { dirname, join } from "path";
+import dateFormat from "dateformat";
+import * as fs from "fs";
+import { docConfig } from "./utils";
+
+async function openDailyNoteFor(date?: Date) {
+  const foamConfiguration = workspace.getConfiguration("foam");
+  const currentDate = date !== undefined ? date : new Date();
+
+  const dailyNotePath = getDailyNotePath(foamConfiguration, currentDate);
+
+  const isNew = await createDailyNoteIfNotExists(
+    foamConfiguration,
+    dailyNotePath,
+    currentDate
+  );
+  await focusDailyNote(dailyNotePath, isNew);
+}
+function getDailyNotePath(configuration: WorkspaceConfiguration, date: Date) {
+  const rootDirectory = workspace.workspaceFolders[0].uri.fsPath;
+  const dailyNoteDirectory: string =
+    configuration.get("openDailyNote.directory") ?? ".";
+  const dailyNoteFilename = getDailyNoteFileName(configuration, date);
+
+  return join(rootDirectory, dailyNoteDirectory, dailyNoteFilename);
+}
+
+function getDailyNoteFileName(
+  configuration: WorkspaceConfiguration,
+  date: Date
+): string {
+  const filenameFormat: string = configuration.get(
+    "openDailyNote.filenameFormat"
+  );
+  const fileExtension: string = configuration.get(
+    "openDailyNote.fileExtension"
+  );
+
+  return `${dateFormat(date, filenameFormat, false)}.${fileExtension}`;
+}
+
+async function createDailyNoteIfNotExists(
+  configuration: WorkspaceConfiguration,
+  dailyNotePath: string,
+  currentDate: Date
+) {
+  if (await pathExists(dailyNotePath)) {
+    return false;
+  }
+
+  await createDailyNoteDirectoryIfNotExists(dailyNotePath);
+
+  const titleFormat: string =
+    configuration.get("openDailyNote.titleFormat") ??
+    configuration.get("openDailyNote.filenameFormat");
+
+  await fs.promises.writeFile(
+    dailyNotePath,
+    `# ${dateFormat(currentDate, titleFormat, false)}${docConfig.eol}${
+      docConfig.eol
+    }`
+  );
+
+  return true;
+}
+
+async function createDailyNoteDirectoryIfNotExists(dailyNotePath: string) {
+  const dailyNoteDirectory = dirname(dailyNotePath);
+
+  if (!(await pathExists(dailyNoteDirectory))) {
+    await fs.promises.mkdir(dailyNoteDirectory, { recursive: true });
+  }
+}
+
+async function focusDailyNote(dailyNotePath: string, isNewNote: boolean) {
+  const document = await workspace.openTextDocument(Uri.file(dailyNotePath));
+  const editor = await window.showTextDocument(document);
+
+  // Move the cursor to end of the file
+  if (isNewNote) {
```

Comment:
iiuc this will move the cursor to the end of the of file if it is a new note:
- shouldn't this happen when the note is _not_ new?
- or, maybe this should just always happen? (if the note is new the beginning and end of file will match, so nothing material changes, but the code is cleaner)

Let me know if I am misunderstanding something

Reply:
I wonder if the reasoning is that, if you've already opened the file before, then you'll have moved the cursor. An example:

I'm editing a specific heading halfway down the page in today's daily note. Now I jump out to have a look at a related thought. When I jump back in (using the Focus Daily Note command) I don't want to jump to the end of the file, I'm expecting the cursor to be where it was when I left?

I only moved this code, so I don't know if that was the intent, but that strikes me as plausible

Reply:
yeah, that makes a lot of sense!